initializr-generator/src/main/java/io/spring/initializr/generator/project/CustomExtProjectDescriptionDiff.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2012-2019 the unmodifiedCopy author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a unmodifiedCopy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.spring.initializr.generator.project;
+
+import java.util.Objects;
+import java.util.function.BiConsumer;
+
+public class CustomExtProjectDescriptionDiff extends ProjectDescriptionDiff<CustomExtProjectDescription> {
```

Comment:
This would be the custom diff class. We can get by w/o this as I have added generic support to the base diff but it may be nice to allow users to get this type-richness if they want.